jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -78,68 +88,47 @@ public static boolean getDefaultUseCaches()
         return __defaultUseCaches;
     }
 
-    /**
-     * Construct a resource from a uri.
-     *
-     * @param uri A URI.
-     * @return A Resource object.
-     * @throws MalformedURLException Problem accessing URI
-     */
-    public static Resource newResource(URI uri)
-        throws MalformedURLException
+    public static Resource.Mount newJarResource(String resource) throws IOException
     {
-        return newResource(uri.toURL());
+        URI uri = URI.create(resource);
+        // If the URI has no scheme, we consider the string actually was a path.
+        if (uri.getScheme() == null)
+            uri = Paths.get(resource).toUri();
+        return newJarResource(uri);
     }
 
-    /**
-     * Construct a resource from a url.
-     *
-     * @param url A URL.
-     * @return A Resource object.
-     */
-    public static Resource newResource(URL url)
+    public static Resource.Mount newJarResource(URI uri) throws IOException
```

Comment:
This method takes a URI to a resource within a jar/file, whilst the `newJarResource(Path)` method takes a path to the war file itself.

I actually think both should be renamed to `mount` and the argument should always be that of the war file (i.e. without "jar:" and "!/").

They should also be moved to the `JarResource` class (if I can convince you to create that class).

Reply:
The `newJarResource(Path)` can easily be a full path deep in a JAR too.

aka: `jar:file:/path/to/file.jar!/entry/in/jar/example.txt`